exchanges/orderbook/depth_test.go

```diff
@@ -712,1689 +504,407 @@ func TestPublish(t *testing.T) {
-	if err != nil {
-		t.Fatal(err)
-	}
-	imbalance, err := depth.GetImbalance()
-	if !errors.Is(err, nil) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, nil)
-	}
-
-	if imbalance != 0 {
-		t.Fatalf("received: '%v' but expected: '%v'", imbalance, 0)
-	}
-}
-
-func TestGetTranches(t *testing.T) {
-	t.Parallel()
-	_, _, err := getInvalidDepth().GetTranches(0)
-	if !errors.Is(err, ErrOrderbookInvalid) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, ErrOrderbookInvalid)
-	}
-
-	depth := NewDepth(id)
-
-	_, _, err = depth.GetTranches(-1)
-	if !errors.Is(err, errInvalidBookDepth) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, errInvalidBookDepth)
-	}
-
-	askT, bidT, err := depth.GetTranches(0)
-	if !errors.Is(err, nil) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, nil)
-	}
-
-	if len(askT) != 0 {
-		t.Fatalf("received: '%v' but expected: '%v'", len(askT), 0)
-	}
-
-	if len(bidT) != 0 {
-		t.Fatalf("received: '%v' but expected: '%v'", len(bidT), 0)
-	}
-
-	err = depth.LoadSnapshot(bid, ask, 0, time.Now(), true)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	askT, bidT, err = depth.GetTranches(0)
-	if !errors.Is(err, nil) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, nil)
-	}
-
-	if len(askT) != 20 {
-		t.Fatalf("received: '%v' but expected: '%v'", len(askT), 20)
-	}
-
-	if len(bidT) != 20 {
-		t.Fatalf("received: '%v' but expected: '%v'", len(bidT), 20)
-	}
-
-	askT, bidT, err = depth.GetTranches(5)
-	if !errors.Is(err, nil) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, nil)
-	}
-
-	if len(askT) != 5 {
-		t.Fatalf("received: '%v' but expected: '%v'", len(askT), 5)
-	}
-
-	if len(bidT) != 5 {
-		t.Fatalf("received: '%v' but expected: '%v'", len(bidT), 5)
-	}
+	askT, bidT, err = depth.GetTranches(5)
+	assert.NoError(t, err, "GetTranches should not error")
+	assert.Len(t, askT, 5, "asks should have correct number of tranches")
+	assert.Len(t, bidT, 5, "bids should have correct number of tranches")
 }
 
 func TestGetPair(t *testing.T) {
 	t.Parallel()
 	depth := NewDepth(id)
 
 	_, err := depth.GetPair()
-	if !errors.Is(err, currency.ErrCurrencyPairEmpty) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, currency.ErrCurrencyPairEmpty)
-	}
+	assert.ErrorIs(t, err, currency.ErrCurrencyPairEmpty, "GetPair should error correctly")
 
 	expected := currency.NewPair(currency.BTC, currency.WABI)
 	depth.pair = expected
 
 	pair, err := depth.GetPair()
-	if !errors.Is(err, nil) {
-		t.Fatalf("received: '%v' but expected: '%v'", err, nil)
-	}
+	assert.NoError(t, err, "GetPair should not error")
 
-	if !pair.Equal(expected) {
-		t.Fatalf("received: '%v' but expected: '%v'", pair, expected)
-	}
+	assert.Equal(t, expected, pair, "GetPair should return correct pair")
 }
 
 func getInvalidDepth() *Depth {
 	depth := NewDepth(id)
 	_ = depth.Invalidate(errors.New("invalid reasoning"))
 	return depth
 }
+
+type movementTest struct {
+	inputs []any
+	expect Movement
+}
+
+var zero = accuracy10dp
```

Comment:
```suggestion
var zero float64
```

Reply:
That would not work. I'll add a comment to clarify.

Reply:
Clarified 3ae88d8a